stan/math/prim/prob/wiener_full_lpdf.hpp

```diff
@@ -0,0 +1,618 @@
+#ifndef STAN_MATH_PRIM_PROB_WIENER_FULL_LPDF_HPP
+#define STAN_MATH_PRIM_PROB_WIENER_FULL_LPDF_HPP
+
+#include <stan/math/prim/fun.hpp>
+#include <stan/math/prim/functor/hcubature.hpp>
+#include <stan/math/prim/prob/wiener5_lpdf.hpp>
+
+namespace stan {
+namespace math {
+namespace internal {
+
+/**
+ * Calculate the derivative of the wiener7 density w.r.t. 'sw'
+ *
+ * @tparam T_y type of scalar variable
+ * @tparam T_a type of boundary separation
+ * @tparam T_v type of drift rate
+ * @tparam T_w type of relative starting point
+ * @tparam T_sv type of inter-trial variability in v
+ * @tparam T_sw type of inter-trial variability in w
+ * @tparam T_err type of log error tolerance
+ *
+ * @param y A scalar variable; the reaction time in seconds
+ * @param a The boundary separation
+ * @param v The drift rate
+ * @param w The relative starting point
+ * @param sv The inter-trial variability of the drift rate
+ * @param sw The inter-trial variability of the relative starting point
+ * @param log_error The log error tolerance
+ * @return Gradient w.r.t. sw
+ */
+template <typename T_y, typename T_a, typename T_v, typename T_w, typename T_sv,
+          typename T_sw, typename T_err>
+inline auto wiener7_grad_sw(const T_y& y, const T_a& a, const T_v& v,
+                            const T_w& w, const T_sv& sv, const T_sw& sw,
+                            T_err log_error) {
+  auto low = w - sw / 2.0;
+  low = (0 > low) ? 0 : low;
```

Comment:
Just want to check. This is doing `w - (sw / 2.0)`. Is that what you are looking for here? 

If `w - (sw / 2.0)` is intended then this is going to cause a flat spot in the gradient calculation anywhere that `w < sw / 2.0`. Same for high

Reply:
Yes, this is what we are looking for at this point.

The described cases will never be true as we check for them and return error messages (see lines 410-422)
```
    if (unlikely(w_vec[i] - .5 * sw_vec[i] <= 0)) {
      [&]() STAN_COLD_PATH {
        std::stringstream msg;
        msg << ", but must be smaller than 2*(A-priori bias) = "
            << 2 * w_vec[i];
        std::string msg_str(msg.str());
        throw_domain_error(function_name,
                           "Inter-trial variability in A-priori bias",
                           sw_vec[i], " = ", msg_str.c_str());
      }();
    }
    if (unlikely(w_vec[i] + .5 * sw_vec[i] >= 1)) {
      [&]() STAN_COLD_PATH {
        std::stringstream msg;
        msg << ", but must be smaller than 2*(1-A-priori bias) = "
            << 2 * (1 - w_vec[i]);
        std::string msg_str(msg.str());
        throw_domain_error(function_name,
                           "Inter-trial variability in A-priori bias",
                           sw_vec[i], " = ", msg_str.c_str());
      }();
    }
```